app/pipeline/models/llm.py

```diff
@@ -0,0 +1,266 @@
+import os
+import typing
+from abc import ABC
+
+from app.model.document import DocumentEdit, Token, Mention
+from app.model.schema import Schema
+from app.model.settings import GptModel
+from app.util.llm_util import get_prediction, extract_json
+
+
+class LLM(ABC):
+    temperature = 0.0
+    model: GptModel = GptModel.GPT_4O_MINI
+    open_ai_key: str
+
+    def __init__(self, model: GptModel = GptModel.GPT_4O_MINI, temperature=0.0):
+        self.model = model
+        self.temperature = temperature
+
+        self.open_ai_key = os.getenv("OPENAI_API_KEY")
+        if self.open_ai_key is None:
+            raise EnvironmentError("OpenAI API key not set")
+
+    @staticmethod
+    def get_schema_mention_string(schema: Schema, for_entities: bool = False) -> str:
+        schema_mentions = schema.schema_mentions
+        if for_entities:
+            schema_mentions = filter(
+                lambda mention: mention.has_entities, schema_mentions
+            )
+
+        schema_mention_string = ""
+        for schema_mention in schema_mentions:
```

Comment:
könnte durch .join vereinfacht werden: 
z.B.:
return "\n".join(
            f'"{schema_mention.tag}": {schema_mention.description}'
            for schema_mention in schema_mentions
        )